src/reducers/filterReducer.js

```diff
@@ -2,6 +2,9 @@ import * as filterActions from 'actions/filterActions'
 
 export const filter = (defaultFilter) => (state = defaultFilter, action) => {
   switch (action.type) {
+    case filterActions.CLEAR_EVENT_FILTER_INCIDENTID:
+      const {incidentId, ...newFilter} = action.filter
+      return {...state, ...newFilter}
```

Comment:
Since incidentId is already part of state, this will just add incidentId back in. You can just do:

return newFilter

Reply:
no it won't, this incidentId is a property of filter, not the incidentId in the state level.